src/FocusApp.Client/Views/Shop/ShopItemPopupInterface.cs

```diff
@@ -208,20 +222,30 @@ private async void PurchaseItem(object sender, EventArgs e)
                     }
 
                     // Add the user's new furniture to the local database
-                    _localContext.UserFurniture.Add(new UserFurniture
+                    user = _localContext.Users.First(u => u.Id == _authenticationService.CurrentUser.Id);
```

Comment:
Here's another example of a Db call that would benefit from being inside of a try-catch block and from using async-await syntax